src/main/navigation/navigation_fixedwing.c

```diff
@@ -463,7 +463,21 @@ int16_t applyFixedWingMinSpeedController(timeUs_t currentTimeUs)
 
 int16_t fixedWingPitchToThrottleCorrection(int16_t pitch)
 {
-    return DECIDEGREES_TO_DEGREES(pitch) * navConfig()->fw.pitch_to_throttle;
+    // Calculate base throttle correction from pitch moving average
+    const int16_t movingAverageCycles = 128; //Number of main loop cycles for average calculation
+    static int16_t averagePitch = (averagePitch * movingAverageCycles + pitch - averagePitch) / movingAverageCycles;
+    const int16_t baseThrottleCorrection = DECIDEGREES_TO_DEGREES(averagePitch) * navConfig()->fw.pitch_to_throttle;
+
+    // Calculate final throttle correction
+    if (pitch > (averagePitch + navConfig()->fw.pitch_to_throttle_thresh)) {
+        return baseThrottleCorrection + DECIDEGREES_TO_DEGREES(pitch - averagePitch - navConfig()->fw.pitch_to_throttle_thresh) * navConfig()->fw.pitch_to_throttle;
+    }
+    else if (pitch < (averagePitch - navConfig()->fw.pitch_to_throttle_thresh)) {
+        return baseThrottleCorrection + DECIDEGREES_TO_DEGREES(pitch - averagePitch + navConfig()->fw.pitch_to_throttle_thresh) * navConfig()->fw.pitch_to_throttle;
+    }
+    else {
+        return baseThrottleCorrection;
+    }
```

Comment:
Not entirely sure what this block does. Is it to apply an extra throttle correction based on the difference between `pitch` and `averagePitch`? Say the airplane is pitched up 7 degrees (`pitch=-70`), with the average pitch being up 3 degrees (`averagePitch=-30`), then this subtracts 70 us from the throttle signal calculated above? Or take the case when `nav_fw_pitch2thr_threshold=0`, then this adds (`pitch` - `averagePitch`)*`fw.pitch_to_throttle` to the throttle value.

Can you explain the goal of this a bit?

Reply:
This section creates a deadband (+/-`nav_fw_pitch2thr_threshold`) around `averagePitch`, where the throttle correction will be kept stable at `baseThrottleCorrection`. By setting `nav_fw_pitch2thr_threshold=0`, you basically disable that deadband.

Reply:
Just to make sure I understand, the idea is that `averagePitch` is equal or close to the target pitch angle (zero for level flight, etc), and at that level of pitch throttle is calculated similar to how it was before this PR. Deviations from that average larger than `fw.pitch_to_throttle_thresh` cause an additional throttle correction. Is this second correction to allow for a more direct response to sudden required pitch changes instead of waiting until `averagePitch` picks up the change?

Just thinking through what happens when you are flying level (`pitch`=`averagePitch`=0) and suddenly get into sinking air. You lose some altitude and the airplane pitches up to correct. If only a small amount of pitch up is required throttle stays at cruise level, because `averagePitch` has barely changed. If more pitch up is needed throttle is immediately increased to avoid losing air speed. When you keep flying through sinking air the airplane will continue to pitch up to maintain altitude so that `averagePitch` accumulates all the pitch angle so `averagePitch`=`pitch` and throttle is managed as in the level flight case. Correct?

It sounds very clever, but I think this needs to be tested in various conditions to avoid overlooking some edge case where this behavior can cause problems. As long is as `movingAverageCycles` is not too large compared to the update frequency of `fixedWingPitchToThrottleCorrection` (no idea what that is btw) I don't foresee and issues. I would like to experiment with some longer moving average windows such as 1 second to see how that behaves.

Awesome work!

Reply:
> Just to make sure I understand, the idea is that averagePitch is equal or close to the target pitch angle (zero for level flight, etc), and at that level of pitch throttle is calculated similar to how it was before this PR. Deviations from that average larger than fw.pitch_to_throttle_thresh cause an additional throttle correction. Is this second correction to allow for a more direct response to sudden required pitch changes instead of waiting until averagePitch picks up the change?

Correct!

>Just thinking through what happens when you are flying level (pitch=averagePitch=0) and suddenly get into sinking air. You lose some altitude and the airplane pitches up to correct. If only a small amount of pitch up is required throttle stays at cruise level, because averagePitch has barely changed. If more pitch up is needed throttle is immediately increased to avoid losing air speed. When you keep flying through sinking air the airplane will continue to pitch up to maintain altitude so that averagePitch accumulates all the pitch angle so averagePitch=pitch and throttle is managed as in the level flight case. Correct?

Correct, and this also applies to the case where you have an altitude change between two waypoints.

>I would like to experiment with some longer moving average windows such as 1 second to see how that behaves.

I'm even thinking 2 to 5 seconds and then tune `nav_fw_pitch2thr_threshold` to get a balance between transition response and smooth throttle in between.